monkey/infection_monkey/exploit/tools/http_tools.py

```diff
@@ -60,3 +69,22 @@ def create_locked_transfer(host, src_path, local_ip=None, local_port=None):
         httpd.start()
         lock.acquire()
         return "http://%s:%s/%s" % (local_ip, local_port, urllib.quote(os.path.basename(src_path))), httpd
+
+
+class MonkeyHTTPServer(HTTPTools):
+    def __init__(self, host):
+        super(MonkeyHTTPServer, self).__init__()
+        self.http_path = None
+        self.http_thread = None
+        self.host = host
+
+    def start(self):
+        # Get monkey exe for host and it's path
+        src_path = try_get_target_monkey(self.host)
+        self.http_path, self.http_thread = MonkeyHTTPServer.try_create_locked_transfer(self.host, src_path)
+
+    def stop(self):
+        if not self.http_path or not self.http_thread:
+            raise Exception("Can't stop http server that wasn't started!")
```

Comment:
Too broad exception clause

Reply:
How to solve it? Create my own exception of http server not started?

Reply:
Yes, I think a custom exception is OK - you can also raise RuntimeError